### PR TITLE
linker: fix various ram/rom iterable section

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

File: cmake/linker_script/common/common-ram.cmake [hunk 1]
```diff
@@ -115,3 +116,36 @@ if(CONFIG_ZTEST_NEW_API)
...
+if(CONFIG_LOG)
+  zephyr_iterable_section(NAME log_mpsc_pbuf GROUP DATA_REGION ${XIP_ALIGN_WITH_INPUT} SUBALIGN 4)
+  zephyr_iterable_section(NAME log_msg_ptr GROUP DATA_REGION ${XIP_ALIGN_WITH_INPUT} SUBALIGN 4)
+endif()
+
+if(CONFIG_PCIE)
+  zephyr_iterable_section(NAME pcie_dev GROUP DATA_REGION ${XIP_ALIGN_WITH_INPUT} SUBALIGN 4)
+endif()
+
+if(CONFIG_USB_DEVICE_STACK OR CONFIG_USB_DEVICE_STACK_NEXT)
```
Comment (nordicjm) [inline cmake/linker_script/common/common-ram.cmake:142]: I see this in the linker file:
```
#if defined(CONFIG_USB_DEVICE_STACK)
	ITERABLE_SECTION_RAM(usb_cfg_data, 4)
```
Not sure if the `CONFIG_USB_DEVICE_STACK_NEXT` part applies - @jfischer-no ?

File: cmake/linker_script/common/common-rom.cmake [hunk 2]
```diff
@@ -193,3 +193,23 @@ zephyr_linker_section_configure(SECTION device_handles INPUT .__device_handles_p
...
+if (CONFIG_HTTP_SERVER)
+  zephyr_iterable_section(NAME http_service_desc KVMA RAM_REGION GROUP RODATA_REGION SUBALIGN 4)
+endif()
+
+if(CONFIG_INPUT)
+  zephyr_iterable_section(NAME input_listener KVMA RAM_REGION GROUP RODATA_REGION SUBALIGN 4)
+endif()
+
+if(CONFIG_USB_DEVICE_STACK OR CONFIG_USB_DEVICE_STACK_NEXT)
+  zephyr_iterable_section(NAME usbd_msc_lun KVMA RAM_REGION GROUP RODATA_REGION SUBALIGN 4)
```
Comment (nordicjm) [inline cmake/linker_script/common/common-rom.cmake:212]: This should be `if(CONFIG_USBD_MSC_CLASS)`

**Review by jfischer-no — DISMISSED** [review]
These changes make no sense when reading [this documentation](https://docs.zephyrproject.org/latest/kernel/iterable_sections/index.html#iterable-sections), so linker snippet support for clang should be fixed.

File: cmake/linker_script/common/common-rom.cmake [hunk 3]
```diff
@@ -200,6 +197,7 @@ endif()
 
 if (CONFIG_LOG)
   zephyr_iterable_section(NAME log_link KVMA RAM_REGION GROUP RODATA_REGION SUBALIGN 4)
+  zephyr_iterable_section(NAME log_backend KVMA RAM_REGION GROUP RODATA_REGION SUBALIGN 4)
```
[the remark this answers is not in the document]
Comment (galak) [reply cmake/linker_script/common/common-rom.cmake:200]: > These changes make no sense when reading [this documentation](https://docs.zephyrproject.org/latest/kernel/iterable_sections/index.html#iterable-sections), so linker snippet support for clang should be fixed.

I don't think that doc takes into account what is needed for generating linker scripts for non-gnu style linkers like arm-clang.  So this is the correct way to handle that for now.

[the remark this answers is not in the document]
Comment (galak) [reply cmake/linker_script/common/common-rom.cmake:200]: > Order is flipped in comparison with .ld file (if it matters)

The order in this file doesn't mater.

File: cmake/linker_script/common/common-rom.cmake (hunk 2 above)
Comment (galak) [reply cmake/linker_script/common/common-rom.cmake:212]: looks like USBD_MSC_CLASS is gone, so maybe I should just remove this?

Comment (galak) [reply cmake/linker_script/common/common-rom.cmake:212]: Done.

**Review by nordicjm — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**